### ISSUE TITLE
GraphicsMagick directory name in the macOS bundle

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#1967]
To create a standalone macOS bundle, we need to pack graphicsmagick and it's config files in the bundle. The config files is located in the directory `GMT/share/GraphicsMagick-1.3.33/config`. The version number (i.e. 1.3.33) may cause some troubles, as @PaulWessel will upgrade to newer versions in the future and users have to change their `MAGICK_CONFIGURE_PATH` when they upgrade to newer GMT versions.

It's better to have the config files in `GMT/share/GraphicsMagick/config`.

**Comment by PaulWessel** [comment]
Yes I agree, I thought about that too with the stupid -1.3.33 line.  All this shit just makes the build-* scripts more complicated to deal with one-off problems, but we might as well. Will look.

**Comment by PaulWessel** [comment]
Hm, I may have to add a stupid link /opt/local/lib/GraphicsMagick to /opt/local/lib/GraphicsMagick-1.3.33 so that I can place that directory in the bundle.  But that means I will forget to update the link.
Currently we have this special install:

```
install (DIRECTORY
	/opt/local/lib/GraphicsMagick-1.3.33
	DESTINATION \${GMT_LIBDIR}
	COMPONENT Runtime)
```

which installs the GraphicsMagick-1.3.33 as such under ${GMT_LIBDIR}.  I don't know how to rename that once there.

**Comment by PaulWessel** [comment]
Maybe add some execute_process to rename that directory?  We define GMT_CONFIG_GM_MESSAGE which looks like yes "(1.3.33").  Perhaps we could break that part down further to have a GMT_CONFIG_GM_VERSION 1.3.33 so we can use this to rename the directory from whatever version it is to the unversioned name?

**Comment by seisman** [comment]
https://cmake.org/pipermail/cmake/2009-December/033850.html

Could you try 
```
install (DIRECTORY
	/opt/local/lib/GraphicsMagick-1.3.33
	DESTINATION \${GMT_LIBDIR}/GraphicsMagick
	COMPONENT Runtime)
```

**Comment by PaulWessel** [comment]
I think I have tried that already and it will simply make
\${GMT_LIBDIR}/GraphicsMagick/GraphicsMagick-1.3.33
But I did not write this down so not 99% sure.

**Comment by PaulWessel** [comment]
I will start with adding a few new cmake variables:

GMT_CONFIG_GM_VERSION
GMT_CONFIG_GS_VERSION
 
in addition to the MESSAGEs

**Comment by PaulWessel** [comment]
There is
RENAME
Specify a name for an installed file that may be different from the original file. Renaming is allowed only when a single file is installed by the command.

but seems to only apply to a single file, not directory.

**Comment by PaulWessel** [comment]
OK, I seeing the subtleties of your piper mail link - will try.

**Comment by PaulWessel** [comment]
sorry, install (DIRECTORY
	/opt/local/lib/GraphicsMagick-1.3.33/
	DESTINATION ${GMT_LIBDIR}/GraphicsMagick
did not work.  Will try

 install (DIRECTORY
	/opt/local/lib/GraphicsMagick-1.3.33/config
	DESTINATION ${GMT_LIBDIR}/GraphicsMagick

**Comment by PaulWessel** [comment]
The latter trick worked.

**Comment by seisman** [comment]
Closed by #1968.
